javascript-practice/src/js/utils/api.js

```diff
@@ -21,6 +21,36 @@ const APIHandler = {
     } catch (error) {
       console.error(error);
 
+      showToastify(error.message, 'toastify-danger');
+    }
+  },
+
+  /**
+   * Sends a POST request to the endpoint with the provided product data
+   * @param {string} endpoint - The endpoint to which the request should be sent
+   * @param {Object} product - The product data to be sent
+   */
+  async post(endpoint, product) {
+    try {
+      const res = await fetch(
+        `${API_ROUTES.BASE_URL}/${endpoint}`,
+        {
+          method: 'POST',
+          headers: {
+              'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(product)
+        }
+      );
+
+      if(!res.ok) {
+        throw new Error(`Failed to post data to ${endpoint}`);
+      }
+
+      showToastify('Product added successfully!', 'toastify-success');
```

Comment:
Should create 2 methods for success and error case in `toastify.js` 

```
const Toast = {
  success(message) {
     ....
  } 
  
  error(message) {
  }
```
Then you can use`Toast.success('Product added successfully!')` here